service/controller/resource/tccp/create.go

```diff
@@ -438,7 +438,15 @@ func (r *Resource) newNATGatewayParams(ctx context.Context, cr infrastructurev1a
 	var natRoutes []template.ParamsMainNATGatewayNATRoute
 	for _, az := range cc.Spec.TenantCluster.TCCP.AvailabilityZones {
 		if az.Name != key.MasterAvailabilityZone(cr) {
```

Comment:
As I understand we want this for the master AZ only. 

```suggestion
		if az.Name == key.MasterAvailabilityZone(cr) {
```